skbio/io/format/tests/test_stockholm.py

```diff
@@ -0,0 +1,148 @@
+                         positional_metadata={'SS_cons': list('CCCCCHHHHHHHH'
+                                                              'HHHHH..EEEEEE'
+                                                              'EE....EEEEEEE'
+                                                              'EEEH')},
+                         index=['O83071/192-246', 'O31698/88-139',
+                                'O31699/88-139'])
+        self.assertEqual(msa, exp)
+
+    def test_stockholm_minimal(self):
+        fp = get_data_path('stockholm_minimal')
+        msa = _stockholm_to_tabular_msa(fp, constructor=DNA)
+        exp = TabularMSA([DNA('TGTGTCGCAGTTGTCGTTTG')], index=['0235244'])
+        self.assertEqual(msa, exp)
+
+    def test_stockholm_rna(self):
+        fp = get_data_path('stockholm_rna')
+        msa = _stockholm_to_tabular_msa(fp, constructor=RNA)
+        exp = TabularMSA([RNA('AAGGGUUAUUUAUAUACUUU'),
+                          RNA('UGCUAAGAGUGGGGAUGAUU'),
+                          RNA('GCCACAACCGAUUAGAUAGA'),
+                          RNA('UUAGAAACCGAUGGACCGAA')],
+                         metadata={'AC': 'G2134T23', 'ID': 'ARD'},
+                         positional_metadata=(
+                         {'AC_cons': list('GGGACUGGACAUCUAUUCAG')}),
+                         index=['RTC2231', 'RTF2124', 'RTH3322', 'RTB1512'])
+        self.assertEqual(msa, exp)
+
+    def test_stockholm_runon_gf(self):
+        fp = get_data_path('stockholm_runon_gf')
+        msa = _stockholm_to_tabular_msa(fp, constructor=DNA)
+        exp = TabularMSA([DNA('ACTGGTTCAATG')],
+                         metadata={'CC': 'CBS domains are small intracellular'
+                                         ' modules mostly found in 2 or four '
+                                         'copies within a protein.'},
+                         index=['GG1344'])
+        self.assertEqual(msa, exp)
+
+    def test_nonexistent_data_errors(self):
+        fp = get_data_path('stockholm_invalid_nonexistent_gr')
+        with assertRaisesRegex(self, StockholmFormatError,
+                               'Markup line references.*RL1355.'):
+            _stockholm_to_tabular_msa(fp, constructor=RNA)
+        fp = get_data_path('stockholm_invalid_nonexistent_gs')
+        with assertRaisesRegex(self, StockholmFormatError,
+                               'Markup line references.*AC14.'):
+            _stockholm_to_tabular_msa(fp, constructor=RNA)
+
+    def test_duplicate_data_error(self):
+        fp = get_data_path('stockholm_duplicate_data')
+        with assertRaisesRegex(self, StockholmFormatError,
+                               'Found multiple data.*ASR132'):
+            _stockholm_to_tabular_msa(fp, constructor=RNA)
+
+    def test_no_data_error(self):
+        fp = get_data_path('empty')
+        with assertRaisesRegex(self, StockholmFormatError, 'No data present'):
+            _stockholm_to_tabular_msa(fp)
+        fp = get_data_path('stockholm_no_data')
```

Comment:
I think the reader should support an empty stockholm file since it can be loaded into an empty `TabularMSA` object. Can you add tests for loading a file with no sequences or metadata:

```
# STOCKHOLM 1.0
//
```

and a file without sequences but with GF metadata?